### DISCUSSION TITLE
image resolution

### DISCUSSION THREAD

**Discussion by eb99** [tomshaffner/PiThermalCam#8] [Q&A] [answered]
HI,

first off, thanks so much for the information you have put together so far, it is fantastic.
now I am fairly inexperienced with this sort of project setup on Pi, so forgive me if I ask silly questions.
I am planning to make this into a portable device, and use components I already have.
I would like to make this work full screen on a waveshare 3.5" display, so I need to reduce the image resolution and display it full screen.
I tried editing the .py file in text editor to change the res, but no change when I run the code.
any tips?
I also want to hardwire some buttons to spare GPIO pins to control the keyboard controls, which I will play with later with help from this page.
https://learn.adafruit.com/retro-gaming-with-raspberry-pi/adding-controls-hardware
future plans would be to have it fast boot, and auto run the thermal cam, and to include a shutdown button.
If anyone can assist with these extra tasks that would be really helpful.
I will work on this myself but it will be very slow given my experience and extreme limited time.
I thought maybe some collaboration on here may be good, as I'm sure others would want to build the same stand alone unit I am trying to build.
Thanks heaps.

**Comment by tomshaffner** [comment]
Which version of the program are you running?

Or put another way, how do you start the program right now?

**Reply by eb99** [reply]
for now I have cloned the repo and open the examples folder and double click the live_video.py and click run when the python editor opens.
I have edited the pi_therm_cam.py image width and height numbers.
I have yet to learn how to autorun python scripts, but I'll get to that later.
thanks

**Reply by eb99** [reply]
OK so I started over and have my waveshare 3.5" LCD working at a more appropriate resolution.
but I am having trouble with the install and running of the PiThermalCam.
I have done the library installation. but when I go to test it I get errors.
"import pithermalcam as ptc
-bash: import: command not found"

I'm not sure what to do from here.

I try running the example Live_video.py and get errors here too.
Python 3.7.3 (/usr/bin/python3)
>>> %cd /home/pi/PiThermalCam/examples
>>> %Run live_video.py
Traceback (most recent call last):
  File "/home/pi/PiThermalCam/examples/live_video.py", line 4, in <module>
    ptc.display_camera_live()
  File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/__init__.py", line 30, in display_camera_live
    thermcam = pithermalcam(output_folder=output_folder)  # Instantiate class
  File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 50, in __init__
    self.update_image_frame()
  File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 227, in update_image_frame
    self._pull_raw_image()
  File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 88, in _pull_raw_image
    self.mlx.getFrame(self._raw_image)  # read mlx90640
  File "/home/pi/.local/lib/python3.7/site-packages/adafruit_mlx90640.py", line 126, in getFrame
    status = self._GetFrameData(mlx90640Frame)
  File "/home/pi/.local/lib/python3.7/site-packages/adafruit_mlx90640.py", line 155, in _GetFrameData
    raise RuntimeError("Too many retries")
RuntimeError: Too many retries

what do I do about this?

sorry lots of dumb questions, I'm out of my depth here!
any help appreciated.
Thanks

**Comment by tomshaffner** [comment, marked as answer]
That error generally means your speed is wrong. Check the documentation section on changing the baudrate.
 
 
  Am Mi., Juli 21, 2021 at 1:49 schrieb ***@***.***>:   


OK so I started over and have my waveshare 3.5" LCD working at a more appropriate resolution.
but I am having trouble with the install and running of the PiThermalCam.
I have done the library installation. but when I go to test it I get errors.
"import pithermalcam as ptc
-bash: import: command not found"

I'm not sure what to do from here.

I try running the example Live_video.py and get errors here too.
Python 3.7.3 (/usr/bin/python3)




%cd /home/pi/PiThermalCam/examples
%Run live_video.py
Traceback (most recent call last):
File "/home/pi/PiThermalCam/examples/live_video.py", line 4, in 
ptc.display_camera_live()
File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/init.py", line 30, in display_camera_live
thermcam = pithermalcam(output_folder=output_folder) # Instantiate class
File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 50, in init
self.update_image_frame()
File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 227, in update_image_frame
self._pull_raw_image()
File "/home/pi/.local/lib/python3.7/site-packages/pithermalcam/pi_therm_cam.py", line 88, in _pull_raw_image
self.mlx.getFrame(self._raw_image) # read mlx90640
File "/home/pi/.local/lib/python3.7/site-packages/adafruit_mlx90640.py", line 126, in getFrame
status = self._GetFrameData(mlx90640Frame)
File "/home/pi/.local/lib/python3.7/site-packages/adafruit_mlx90640.py", line 155, in _GetFrameData
raise RuntimeError("Too many retries")
RuntimeError: Too many retries




what do I do about this?

sorry lots of dumb questions, I'm out of my depth here!
any help appreciated.
Thanks

**Reply by eb99** [reply]
Thanks heaps for that. speed was wrong, return line change in the middle of the code didn't help.
I am still not able to run it from the command line as in the instructions, it will however work inside Thonny python editor.
Any thoughts on how to script it to start given the error I am getting?
Thanks again,

**Comment by tomshaffner** [comment]
You need to run it in python. If you're importing it on bash, type python first.

-----Original Message-----

**Reply by eb99** [reply]
ok so I have it working at the resolution I want, I've edited the on screen text so its readable on the tiny screen, multiline text, and made the image fullscreen, and all works so far.
I have tried all options on this page to have it auto run, but no dice.
https://www.makeuseof.com/how-to-run-a-raspberry-pi-program-script-at-startup/
Any ideas on how to make the shortcut that is on the desktop execute automatically at startup?
Thanks

**Comment by tomshaffner** [comment]
When you tried the cron solution, did the program you called mirror what the shortcut on the screen calls? I.e. did it open a python environment and then call the program from it?
Open up the properties of the shortcut and see what it's doing and then mirror that in cron with a full path for everything.
Similar with the other two approaches. Saying you tried them and they didn't work isn't helpful; why didn't they work? What went wrong? What did you try to debug? What result did that lead to? What error messages or logs resulted?

 
  Am Mi., Aug. 4, 2021 at 1:09 schrieb ***@***.***>:   


ok so I have it working at the resolution I want, I've edited the on screen text so its readable on the tiny screen, multiline text, and made the image fullscreen, and all works so far.
I have tried all options on this page to have it auto run, but no dice.
https://www.makeuseof.com/how-to-run-a-raspberry-pi-program-script-at-startup/
Any ideas on how to make the shortcut that is on the desktop execute automatically at startup?
Thanks

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by eb99** [reply]
HI,

the shortcut is setup with the following:
[Desktop Entry]
Version=1.0
Name=ThermalCam
Icon=xfce4-ui
Path=/home/pi/PiThermalCam/sequential_versions/
Exec=/usr/bin/python3 /home/pi/PiThermalCam/sequential_versions/opencv_therm_cam.py
Terminal=true
Type=Application
StartUpNotify=true
Name[en_US]=ThermalCam
Name[en_AU]=ThermalCam
StartupNotify=true

and when you double click it, a terminal screen opens with the keyboard shortcut list, then the image starts up fullscreen as expected.

when I try either rc.local, crontab, or startup methods, nothing happens on reboot, no terminal screen, no warnings or dialog boxes.

Given my lack of knowledge on this stuff, where do I even look to find logs or debugging info?
I really just need it to double click the desktop icon once booted, but I'm out of ideas on how to to that.
I tried both opions on this page as well, and still no joy.
https://www.tomshardware.com/amp/how-to/run-script-at-boot-raspberry-pi
I thought the bit about includng "Exec=lxterminal --command”/bin/bash -c ‘sudo python3" might have helped but not.
thanks heaps for your time on this, I am sorry about all the questions.

**Comment by tomshaffner** [comment]
The exec line in the above is the key. Notice how it has the python instance you're calling first, then a space, then the path to the python file. This is what you want to replicate in whichever boot process you're using.
Beyond that, just Google for the boot approach you're using and "python" for guides or help. Maybe add in your device name if there's anything unique about it. Good luck!
 
 
  Am Mi., Aug. 4, 2021 at 21:20 schrieb ***@***.***>:   


HI,

the shortcut is setup with the following:
[Desktop Entry]
Version=1.0
Name=ThermalCam
Icon=xfce4-ui
Path=/home/pi/PiThermalCam/sequential_versions/
Exec=/usr/bin/python3 /home/pi/PiThermalCam/sequential_versions/opencv_therm_cam.py
Terminal=true
Type=Application
StartUpNotify=true
Name[en_US]=ThermalCam
Name[en_AU]=ThermalCam
StartupNotify=true

and when you double click it, a terminal screen opens with the keyboard shortcut list, then the image starts up fullscreen as expected.

when I try either rc.local, crontab, or startup methods, nothing happens on reboot, no terminal screen, no warnings or dialog boxes.

Given my lack of knowledge on this stuff, where do I even look to find logs or debugging info?
thanks heaps for your time on this, I am sorry about all the questions.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by eb99** [reply]
HI,
I'm still getting nowhere with this automated execute.
any other suggestions on methods to automate it?

**Reply by eb99** [reply]
HI,
getting somewhere...slowly.
any idea on what this error means?

Aug 10 11:24:53 raspberrypi rc.local[513]:   File "/home/pi/PiThermalCam/sequential_versions/opencv_therm_cam.py", line 6, in <module>
Aug 10 11:24:53 raspberrypi rc.local[513]:     import time,board,busio
Aug 10 11:24:53 raspberrypi rc.local[513]: ModuleNotFoundError: No module named 'board'


Thonny debug seems to have a problem with this line also, but no suggestion on what to do with it.

import time,board,busio

any ideas how to fix this?
it doesn't seem to stop it working if I double click the icon.

**Reply by eb99** [reply]
ok started again from scratch, followed your instructions exactly.
First problem, will not run from the test commands mentioned 
import pithermalcam as ptc

ptc.test_camera()

or 
import pithermalcam as ptc

ptc.display_camera_live()

if I open it in Thonny, I get the following error if I try to run it.

"Python 3.7.3 (/usr/bin/python3)
>>> %Run opencv_therm_cam.py
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/numpy/core/__init__.py", line 22, in <module>
    from . import multiarray
  File "/home/pi/.local/lib/python3.7/site-packages/numpy/core/multiarray.py", line 12, in <module>
    from . import overrides
  File "/home/pi/.local/lib/python3.7/site-packages/numpy/core/overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: libf77blas.so.3: cannot open shared object file: No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pi/PiThermalCam/sequential_versions/opencv_therm_cam.py", line 7, in <module>
    import numpy as np
  File "/home/pi/.local/lib/python3.7/site-packages/numpy/__init__.py", line 150, in <module>
    from . import core
  File "/home/pi/.local/lib/python3.7/site-packages/numpy/core/__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.7 from "/usr/bin/python3"
  * The NumPy version is: "1.21.1"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: libf77blas.so.3: cannot open shared object file: No such file or directory

>>> 
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Python 3.7.3 (/usr/bin/python3)
>>> 
"

I am completely confused now, and as far as I'm concerned, your instructions no longer work, and I don't know how to resolve this.
